Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * This file is part of the Symfony CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class Php4ConstructorFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        // If the code uses namespaces, then it's certainly not PHP4 ;)
+        if (count($tokens->findGivenKind(T_NAMESPACE))) {
+            return $content;
+        }
+
+        foreach ($tokens->findGivenKind(T_CLASS) as $index => $token) {
+            $classIndex = $tokens->getNonWhitespaceSibling($index, 1);
+            $className = $tokens[$classIndex]->getContent();
+            $classStart = $tokens->getNextTokenOfKind($classIndex, array('{'));
+            $classEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classStart);
+
+            $p4 = $this->findFunction($tokens, $className, $classStart, $classEnd);
+
+            if (!$p4) {
+                // No PHP4-constructor!
+                continue;
+            }
+
+            $p5 = $this->findFunction($tokens, '__construct', $classStart, $classEnd);
+
+            if (!$p5) {
+                // No PHP5-constructor, we can rename the old one
+                $tokens[$p4['name']]->setContent('__construct');
+            } else {
+                // Does PHP4 call __construct?
+                if (stripos($p4['body'], '$this->__construct') !== false) {
```

Comment:
Please use coding standards statement in contributing file.

Yoda with you should be.